controller/src/test/java/io/pravega/controller/task/Stream/StreamMetadataTasksTest.java

```diff
@@ -809,17 +819,82 @@ public void sealStreamTest() throws Exception {
 
         //reseal a sealed stream.
         assertEquals(UpdateStreamStatus.Status.SUCCESS, streamMetadataTasks.sealStream(SCOPE, stream1, null).get());
+        assertTrue(Futures.await(processEvent(requestEventWriter)));
 
         //scale operation on the sealed stream.
         AbstractMap.SimpleEntry<Double, Double> segment3 = new AbstractMap.SimpleEntry<>(0.0, 0.2);
-        AbstractMap.SimpleEntry<Double, Double> segment4 = new AbstractMap.SimpleEntry<>(0.3, 0.4);
+        AbstractMap.SimpleEntry<Double, Double> segment4 = new AbstractMap.SimpleEntry<>(0.2, 0.4);
         AbstractMap.SimpleEntry<Double, Double> segment5 = new AbstractMap.SimpleEntry<>(0.4, 0.5);
 
         ScaleResponse scaleOpResult = streamMetadataTasks.manualScale(SCOPE, stream1, Collections.singletonList(0),
                 Arrays.asList(segment3, segment4, segment5), 30, null).get();
 
         // scaling operation fails once a stream is sealed.
         assertEquals(ScaleStreamStatus.FAILURE, scaleOpResult.getStatus());
+
+        AssertExtensions.assertThrows("Scale should not be allowed as stream is already sealed",
```

Comment:
Can't we separate this out into a different test case? Otherwise, the test becomes huge and it is difficult to understand the functionality that is being tested. I'm also wondering whether this looking more like an integration test as it is testing a flow rather than the specific functionality you are adding here.

Reply:
Yes, but it isnt quite integration test. 
Since our tasks can post events back into the stream, we mock the writers and pass it for processing of our tasks and then verify that events are posted along expected lines. 
So in these tests we typically set up some state where we expect event to be posted back. verify that it has been posted back.
then we change the overall state to ensure that when we initiate processing next time, it demonstrate the expected behaviour..

Reply:
Also, separated into different test.